azurelinuxagent/common/utils/networkutil.py

```diff
@@ -171,15 +213,21 @@ def get_iptables_drop_command(wait, command, destination):
     def get_firewalld_accept_command(command, destination, uid, wait=""):
         cmd = AddFirewallRules.__get_firewalld_base_command(command)
         cmd.extend(
-            AddFirewallRules.__get_common_accept_command_params(wait, AddFirewallRules.__ACCEPT_COMMAND, destination,
+            AddFirewallRules.__get_common_accept_command_params(wait, AddFirewallRules.get_append_command(), destination,
                                                                 uid))
         return cmd
 
+    @staticmethod
+    def get_firewalld_accept_dns_tcp_command(command, destination):
+        cmd = AddFirewallRules.__get_firewalld_base_command(command)
+        cmd.extend(['-t', 'security', '-I', 'OUTPUT', '-d', destination, '-p', 'tcp', '--destination-port', '53', '-j', 'ACCEPT'])
```

Comment:
I am not a big fan of the rule duplication for iptables and firewalld. I would prefer if the common code/rules can come from one single function to avoid code duplication as much as possible since it makes modifying it easier in the future if needed. 

Is there anyway that duplication can be avoided here too?

Reply:
I have modified the logic a bit now with one optional parameter for firewalld . For firewalld we need to specify if its a "--query-passthrough " or "--passthrough" . if it is not specified then I generate the iptable command. 

Example:  

firewall-cmd --permanent --direct **--query-passthrough** ipv4 _-t security _-I OUTPUT -d 168.63.129.16 -p tcp --destination-port 53 -j ACCEPT__

firewall-cmd --permanent --direct **--passthrough** ipv4 _-t security _-I OUTPUT -d 168.63.129.16 -p tcp --destination-port 53 -j ACCEPT__

iptables -w _-t security -I OUTPUT -d 168.63.129.16 -p tcp --destination-port 53 -j ACCEPT_

Please let me know you thoughts on this @narrieta @larohra